apps/angular/src/app/features/anime/anime-table/anime-table.component.ts

```diff
@@ -0,0 +1,99 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { PageEvent } from '@angular/material/paginator';
+import { AnimeService } from '@js-camp/angular/core/services/anime.service';
+import { Anime } from '@js-camp/core/models/anime';
+import { AnimeSortField, GetAnimeParams } from '@js-camp/core/models/get-anime-params';
+import { Pagination } from '@js-camp/core/models/pagination';
+import { BehaviorSubject, Observable, tap, map, debounceTime, switchMap, shareReplay } from 'rxjs';
+
+const defaultParams: GetAnimeParams = {
+	limit: 10,
+	page: 0,
+	sorting: { field: AnimeSortField.None, direction: 'asc' },
+	filters: {
+		type: [],
+		search: '',
+	},
+};
+
+const REQUEST_DEBOUNCE_TIME = 500;
+
+/** Anime table component. */
+@Component({
+	selector: 'camp-anime-table',
+	templateUrl: './anime-table.component.html',
+	styleUrls: ['./anime-table.component.css'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AnimeTableComponent {
+
+	/** Columns in a table. */
+	public readonly displayedColumns = ['image', 'titleEnglish', 'titleJapanese', 'aired.start', 'type', 'status'];
+
+	/** Anime list. */
+	public readonly animeList$: Observable<readonly Anime[]>;
+
+	/** Anime is loading. */
+	public readonly isLoading$ = new BehaviorSubject<boolean>(false);
+
+	/** Anime query parameters. */
+	private readonly params$: Observable<GetAnimeParams>;
+
+	/** Current table page data. */
+	private readonly pagination$: Observable<Pagination<Anime>>;
+
+	/** Current table page. */
+	public readonly page$ = new BehaviorSubject<number>(defaultParams.page);
+
+	/** Number of elements per page. */
+	public limit = defaultParams.limit;
+
+	/** Total number of anime. */
+	public readonly animeTotalCount$: Observable<number>;
+
+	/**
+	 * @param animeService Anime request service.
+	 */
+	public constructor(public readonly animeService: AnimeService) {
```

Comment:
Service shouldn't be public

```suggestion
	public constructor(private readonly animeService: AnimeService) {
```